src/common/CudaDataUtils.hpp

```diff
@@ -42,6 +42,31 @@ __device__ __forceinline__ unsigned long long device_timer()
   return global_timer;
 }
 
+/*!
+ * \brief Method for launching a CUDA kernel with given configuration.
+ *
+ *        Note: method includes a call to check whether number of args
+ *              in kernel signature matches number of args passed to this 
+ *              method.
+ */
+template <typename... Args, typename F = void (*)(Args...)>
+void RPlaunchCudaKernel(F kernel,
+                        const size_t& numBlocks, const size_t& dimBlocks,
+                        std::uint32_t sharedMemBytes, cudaStream_t stream,
+                        Args... args) 
+{
+  constexpr size_t count = sizeof...(Args);
+  auto tup = std::tuple<Args...>{args...};
+  auto chk_tup = checkArgsCount(kernel, tup);
+  void* arg_arr[count];
+  packArgs<0>(chk_tup, arg_arr);
```

Comment:
```suggestion
  void* arg_arr[count] {(void*)&args...};
```
I think you can just do this.

Reply:
That would be simpler. I'll try it. I basically borrowed what was in the ROCM 5.6.1 HIP runtime header file.

Reply:
If their impl is in a macro they might have to use a function to do this. Even then you'd think they'd want param pack expansion if they can use c++11.

Reply:
I think they have both a function and a macro and  they use it in both places.